src/pybamm/solvers/c_solvers/idaklu/IDAKLUSolverOpenMP.inl

```diff
@@ -127,6 +130,26 @@ void IDAKLUSolverOpenMP<ExprSet>::InitializeStorage(int const N) {
           vector<realtype>(length_of_return_vector, 0.0)
       )
   );
+
+  if (save_hermite) {
```

Comment:
I added an initialization for safety (see below). This is not always equal to `solver_opts.hermite_interpolation` because we need to satisfy additional checks that the full solution `y` is available and that we are saving all adaptive steps. If we don't, then we cannot guarantee the interpolation accuracy.

https://github.com/pybamm-team/PyBaMM/blob/a4aa9fe17dceadad6ac70e566663b824c3b78ddf/src/pybamm/solvers/c_solvers/idaklu/IDAKLUSolverOpenMP.inl#L90-L91